courseMaterial/courseMaterial.go

```diff
@@ -0,0 +1,80 @@
+package coursematerial
+
+import (
+	"context"
+	"database/sql"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"learningbay24.de/backend/models"
+)
+
+func CreateMaterial(db *sql.DB, name string, uri string, uploaderid, courseID int, local int8) error {
+
+	// Begin transaction with database handle
+	tx, err := db.BeginTx(context.Background(), nil)
+	if err != nil {
+		return err
+	}
+
+	// Creates File (courseMaterial) struct
+	cm := &models.File{Name: name, URI: uri, UploaderID: uploaderid, Local: local}
```

Comment:
Du greifst auf die bereits vorhandene Methode zu. HIer ist keine Logik zur Speicherung der übergebenen Datei vorhanden: Du brauchst im Parameter noch die bytes, die vom Frontend ans Backend geschickt werden, um die an die bereits vorhandene Methode zum Erstellen einer Datei zu übergeben, damit diese die erstellt und die Datei in der Datenbank anlegt. Mit der zurückgegebenen id kannst du die Datei dann im Kurs verlinken.
https://github.com/LearningBay24/LearningBay24-backend/blob/main/db/file.go#L25=